klippy/extras/pwm_tool.py

```diff
@@ -48,21 +53,30 @@ def setup_start_value(self, start_value, shutdown_value):
         self._start_value = max(0., min(1., start_value))
         self._shutdown_value = max(0., min(1., shutdown_value))
     def _build_config(self):
+        pin_ident_string = f"(pin '{self._pin}' of mcu '{self._mcu._name}')"
         printer = self._mcu.get_printer()
         config_error = printer.config_error
         if self._max_duration and self._start_value != self._shutdown_value:
             raise config_error("Pin with max duration must have start"
-                               " value equal to shutdown value")
+                               " value equal to shutdown value " +
+                               pin_ident_string)
         cmd_queue = self._mcu.alloc_command_queue()
         curtime = printer.get_reactor().monotonic()
         printtime = self._mcu.estimated_print_time(curtime)
         self._last_clock = self._mcu.print_time_to_clock(printtime + 0.200)
         cycle_ticks = self._mcu.seconds_to_clock(self._cycle_time)
-        if cycle_ticks >= 1<<31:
-            raise config_error("PWM pin cycle time too large")
+        max_possible_duration = self.get_max_possible_duration()
+        if self._cycle_time >= max_possible_duration:
+            raise config_error("PWM pin cycle time of "
+                              f"{self._cycle_time}s ({cycle_ticks} ticks) "
+                               "too large " + pin_ident_string)
         self._duration_ticks = self._mcu.seconds_to_clock(self._max_duration)
-        if self._duration_ticks >= 1<<31:
-            raise config_error("PWM pin max duration too large")
+        if self._max_duration > max_possible_duration:
+            raise config_error("PWM pin max duration of " +
+                             f"{self._max_duration}s "
```

Comment:
I hope that using python3-style formatstrings is ok?
I can change that to the other style being used in the rest of the project, but these take more lines of code.

Reply:
I just saw that the CI also tests for python2 compatibility. On it